multivariate_grouped_data_plot.Rmd

```diff
@@ -0,0 +1,825 @@
+# 堆积条形图  y = counts，x = cut，同时柱颜色由变量“color”决定
+# position = position_stack() 
+ggplot(df, aes(x = cut, y = counts)) +
+  geom_bar(
+    aes(color = color, fill = color),
+    stat = "identity", position = position_stack()
+    ) +
+  scale_color_manual(values = c("#0073C2FF", "#EFC000FF"))+
+  scale_fill_manual(values = c("#0073C2FF", "#EFC000FF"))
+# position = position_dodge() 
+p <- ggplot(df, aes(x = cut, y = counts)) +
+  geom_bar(
+    aes(color = color, fill = color),
+    stat = "identity", position = position_dodge(0.8),
+    width = 0.7
+    ) +
+  scale_color_manual(values = c("#0073C2FF", "#EFC000FF"))+
+  scale_fill_manual(values = c("#0073C2FF", "#EFC000FF"))
+p
+```
+
+另外，可以使用ggpubr软件包轻松创建点图。
+
+```{r}
+ggdotchart(df, x = "cut", y ="counts",
+           color = "color", palette = "jco", size = 3, 
+           add = "segment", 
+           add.params = list(color = "lightgray", size = 1.5),
+           position = position_dodge(0.3),
+           ggtheme = theme_pubclean()
+           )
+```
+
+### 2.3 加入数据标签（并列柱状图）
+
+```{r}
+p + geom_text(
+  aes(label = counts, group = color), 
+  position = position_dodge(0.8),
+  vjust = -0.3, size = 3.5
+)
+```
+
+### 2.4 加入数据标签（堆积柱状图）
+
+- 按cut和color列对数据进行排序。当position_stack()反转组顺序时，颜色列应按降序排序。
+
+- 计算每个切割类别的累计计数总和。用作标签的y坐标。要将标签放在条形图的中间，我们将使用cumsum（counts）-0.5 *计数。
+
+- 创建条形图并添加标签。
+
+```{r}
+df <- df %>%
+  arrange(cut, desc(color)) %>%
+  mutate(lab_ypos = cumsum(counts) - 0.5 * counts) 
+head(df, 4)
+```
+```{r}
+ggplot(df, aes(x = cut, y = counts)) +
+  geom_bar(aes(color = color, fill = color), stat = "identity") +
+  geom_text(
+    aes(y = lab_ypos, label = counts, group = color),
+    color = "white"
+  ) + 
+  scale_color_manual(values = c("#0073C2FF", "#EFC000FF"))+
+  scale_fill_manual(values = c("#0073C2FF", "#EFC000FF")) 
+```
+
+### 2.5 其他类型柱状图
+
+可以绘制具有离散x轴和离散y轴的两个离散变量，而不是创建计数的条形图。
+
+每个单独的点均按组显示。对于给定的组，点数与该组中的记录数相对应。
+
+关键函数：geom_jitter()。
+
+主要参数：alpha, color, fill, shape，size。
+
+```{r}
+diamonds.frac <- dplyr::sample_frac(diamonds, 1/5)
+ggplot(diamonds.frac, aes(cut, color)) +
+  geom_jitter(aes(color = cut), size = 0.3)+
+  ggpubr::color_palette("jco")+
+  ggpubr::theme_pubclean()
+```
+
+   
+    
+    
+## 3 连续变量分组制图
+
+在本节中，我们将展示如何使用箱形图、小提琴图、带状图和替代方法绘制成有组别的连续变量。 
+
+我们还将介绍如何自动添加比较组的p值。  
+
+在本节中，我们将theme_bw()设置为默认的ggplot主题：
+
+```{r}
+theme_set(
+  theme_bw()
+)
+```
+
+### 3.1 数据格式
+
+演示数据集：ToothGrowth  
+
+连续变量：len (即牙齿长度), 作为y轴变量。  
+
+分组变量：dose (维生素C的剂量水平：每天0.5、1、2毫克)，作为x轴变量。  
+
+首先，将变量dose从数字转换为离散因子变量：  
+ 
+```{r}
+data("ToothGrowth")
+ToothGrowth$dose <- as.factor(ToothGrowth$dose)
+head(ToothGrowth)
+```
+
+### 3.2 箱型图
+
+关键功能：geom_boxplot()      
+
+关键参数：
+
+- width：箱形图的宽度。    
+
+- notch：逻辑参数。如果为TRUE，则创建一个缺口箱图。凹口在中位数附近显示置信区间，该置信区间通常基于中位数+/- 1.58*IQR/sqrt(n)。槽口用于组与组之间的比较；如果两个盒子的凹口不重叠，则证明中位数不同。  
+
+- color, size, linetype：边框线的颜色，大小和线型。    
+
+- fill：箱形图区域填充颜色。     
+
+- outlier.colour, outlier.shape, outlier.size：异常值点的颜色，形状和大小。   
+
+#### 3.2.1 创建基本箱形图   
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.